iothub_client/inc/iothub_client_diagnostic.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/** @file   iothub_client_diagnostic.h
+*	@brief  The @c diagnostic component.
```

Comment:
That's good that you added this documentation, thanks.
Suggestion: could you add more details to give a better idea of what the module does? 'The diagnostic component' doesn't explain much.

Reply:
good suggestion